crates/accelerate/src/sparse_observable.rs

```diff
@@ -2337,6 +2402,63 @@ impl PySparseObservable {
         Ok(inner.into())
     }
 
+    /// Express the observable in Pauli terms only, by writing each projector as sum of Pauli terms.
+    ///
+    /// .. warning::
+    ///
+    ///     Beware that this will use at least :math:`2^n` terms if there are :math:`n`
+    ///     single-qubit projectors present, which can lead to an exponential number of terms.
+    ///
+    /// Returns:
+    ///     The same observable, but expressed in Pauli terms only.
+    ///
+    /// Examples:
+    ///
+    ///     >>> obs = SparseObservable("+")
+    ///     >>> obs.to_paulis()
+    ///     <SparseObservable with 2 terms on 1 qubit: (0.5+0j)() + (0.5+0j)(X_0)>
+    fn to_paulis(&self) -> PyResult<Self> {
+        let inner = self.inner.read().map_err(|_| InnerReadError)?;
+        Ok(inner.to_paulis()?.into())
+    }
+
+    /// Express the observable in terms of a sparse list format.
+    ///
+    /// This can be seen as counter-operation of :meth:`.SparseObservable.from_sparse_list`, however
+    /// the order of terms is not guaranteed to be the same at after a roundtrip to a sparse
+    /// list and back. Use :meth:`SparseObservable.simplify` to obtain a canonical representation.
+    ///
+    /// Examples:
+    ///     
+    ///     >>> obs = SparseObservable.from_list([("IIXIZ", 2j), ("IIZIX", 2j)])
+    ///     >>> reconstructed = SparseObservable.from_sparse_list(obs.to_sparse_list(), obs.num_qubits)
+    ///     >>> assert obs.simplify() == reconstructed.simplify()
+    #[pyo3(signature = ())]
+    fn to_sparse_list(&self, py: Python) -> PyResult<Py<PyList>> {
+        let inner = self.inner.read().map_err(|_| InnerReadError)?;
+
+        // turn a 3-tuple of (bit terms, indices, coeff) into a Python tuple
+        let to_py_tuple = |bits: &[BitTerm], indices: &[u32], coeff: Complex64| {
+            let mut pauli_string = String::new();
+            for bit in bits {
+                pauli_string.push_str(bit.py_label());
+            }
+            let py_string = PyString::new(py, &pauli_string).unbind();
+            let py_indices = PyList::new(py, indices)?.unbind();
+            let py_coeff = coeff.into_py_any(py)?;
+
+            PyTuple::new(py, vec![py_string.as_any(), py_indices.as_any(), &py_coeff])
+        };
+
+        let sparse_list = inner
+            .iter()
+            .map(|view| to_py_tuple(view.bit_terms, view.indices, view.coeff))
+            .collect::<PyResult<Vec<_>>>()?;
+
+        let out = PyList::new(py, sparse_list)?;
+        Ok(out.unbind())
```

Comment:
Two points here, both pretty minor:

1. why deconstruct the `SparseView` _before_ `to_py_tuple` and not just accept that in the closure?
2. it'd probably be a bit more efficient to construct a `let out = PyList::empty(py)?;` and do `for term in inner.iter(): out.append(to_py_tuple(term)?)?;` than to collect into a `Vec` and then convert that into a list

Reply:
1. that's legacy from when `to_paulis` returned the bits/indices/coeffs separately
2. good idea, I was looking for a way to avoid this!

Reply:
Done in 984a75c54